modules/components/AsyncConnect.js

```diff
@@ -56,7 +53,25 @@ export default class AsyncConnect extends Component {
   }
 
   isLoaded() {
-    return this.context.store.getState().reduxAsyncConnect.loaded;
+    const componentKeys = this.props.components.reduce((componentsMemo, component) => {
+      if (component && component.reduxAsyncConnect && component.reduxAsyncConnect.length) {
+        component.reduxAsyncConnect.forEach(asyncConnectComponent => {
+          if (asyncConnectComponent.key) {
+            componentsMemo.push(asyncConnectComponent.key);
+          }
+        });
+      }
+
+      return componentsMemo;
+    }, []);
+
+    const reduxAsyncConnectComponentsLoaded = componentKeys.reduce((loadedMemo, key) =>
+      loadedMemo &&
+      !!deep(this.context.store.getState(), `reduxAsyncConnect.loadState.${key}.loaded`)
```

Comment:
since we do an iteration over all promises I'd rather not use `deep-get-set`

1. don't `getState()` on every iteration

```js
const state = this.context.store.getState().reduxAsyncConnect.loadState;

if (!state.loaded) {
  return false;
}

const reduxAsyncConnectComponentsLoaded = componentKeys.reduce((loadedMemo, key) =>
  loadedMemo && state[key] && state[key].loaded
, true)

return reduxAsyncConnectComponentsLoaded;
```